botocore/docs/sharedexample.py

```diff
@@ -0,0 +1,207 @@
+import re
+import numbers
+from botocore.utils import parse_timestamp
+from datetime import datetime
+
+
+class SharedExampleDocumenter(object):
+    def document_shared_example(self, example, prefix, section,
+                                operation_model):
+        """Documents a single shared example based on its definition.
+
+        :param example: The model of the example
+
+        :param prefix: The prefix to use in the method example.
+
+        :param section: The section to write to.
+        """
+        section.style.new_paragraph()
+        section.write(example.get('description'))
+        section.style.new_line()
+        self.document_input(section, example, prefix,
+                            operation_model.input_shape)
+        self.document_output(section, example, operation_model.output_shape)
+
+    def document_input(self, section, example, prefix, shape):
+        input_section = section.add_new_section('input')
+        input_section.style.start_codeblock()
+        if prefix is not None:
+            input_section.write(prefix)
+        params = example['input']
+        comments = example.get('comments')
+        if comments:
+            comments = comments.get('input')
+        param_section = input_section.add_new_section('parameters')
+        self._document_params(param_section, params, comments, [], shape)
+        closing_section = input_section.add_new_section('input-close')
+        closing_section.style.new_line()
+        closing_section.style.new_line()
+        closing_section.write('print(response)')
+        closing_section.style.end_codeblock()
+
+    def document_output(self, section, example, shape):
+        output_section = section.add_new_section('output')
+        output_section.writeln('Expected Output:')
+        output_section.style.start_codeblock()
+        params = example.get('output', {})
+
+        # There might not be an output, but we will return metadata anyway
+        params['ResponseMetadata'] = {"...": "..."}
+        comments = example.get('comments')
+        if comments:
+            comments = comments.get('output')
+        self._document_dict(output_section, params, comments, [], shape, True)
+        closing_section = output_section.add_new_section('output-close')
+        closing_section.style.end_codeblock()
+
+    def _document(self, section, value, comments, path, shape):
+        """
+        :param section: The section to add the docs to.
+        :param value: The input / output values representing the parameters that
+                      are included in the example.
+        :param comments: The dictionary containing all the comments to be
+                         applied to the example.
+        :param path: A list describing where the documenter is in traversing the
+                     parameters. This is used to find the equivalent location
+                     in the comments dictionary.
+        """
+        if isinstance(value, dict):
```

Comment:
Based on the logic you assume that all dictionaries will be ``structures``. However, dictionaries can be maps as well, which will affect how you parse the model. I think you will have to update the if statements to account for maps.